samza-core/src/main/java/org/apache/samza/runtime/AbstractApplicationRunner.java

```diff
@@ -106,4 +112,25 @@ final ExecutionPlan getExecutionPlan(StreamApplication app) throws Exception {
   final StreamManager getStreamManager() {
     return streamManager;
   }
+
+  /**
+   * Write the execution plan JSON to a file
+   * @param planJson JSON representation of the plan
+   */
+  final void writePlanJsonFile(String planJson) {
```

Comment:
Right now if we have the file, then user can simplify put the file into the same directory with plan.html and then they can view it. The plan is also serialized in the config so the user can also paste that in a file and view it locally (put the file under the same folder as plan.html). In the future we can also provide a web page that let user enters the json. But to me, the future dashboard will be a better place to put this. Right now generating the file helps local debugging by opening the html directly.